### PR TITLE
JS Capstone Project: Movie Shop

### PR DESCRIPTION
**Pull request by mnaibei**
### What we have done on this project:

- Created config files for the project
- Split the work between ourselves
- Configured web pack
- Installed jest
- Created index and style source files
- Split our code into modules
- Displayed movies to DOM
- Implemented like function
- Created a popup modal
- Used involvement API to record likes and comments
- Fetched likes and comments from API
- Implemented comment function
- Published project to GitHub pages
- Recorded video detailing the description of our project. Link: https://drive.google.com/file/d/1Fo4FlQ5luvMC5IJgNQfHuS5sQ3MAq7C1/view?usp=sharing
- Created PR

### CODE REVIEW

File: src/modules/movies.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    // like button functionality
+    const likeBtn = document.querySelectorAll('.like');
+    likeBtn.forEach((button) => {
+      button.addEventListener('click', async (item) => {
+        const { id } = item.target; // get the ID of the clicked element
+        postLikes(id); // pass the ID to postLikes
+        item.preventDefault();
+      });
+      updateLikesCount();
+    });
```
Comment (amejid) [inline src/modules/movies.js:53]: - [x] You did a great job on almost all parts of the app. :+1: Let's work on some more details to achieve the project requirements. 
When clicking the heart icon, the number of likes is not getting updated automatically. I have to reload the page to see the changes. Kindly make it so that the number updates when clicking the button. ✍️

**Project requirements**
Screenshot from 2023-04-21 16-48-12

File: src/modules/tests/commentInvolvement.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+
+  await fetch(commentApi, options).then((res) => res.text());
+
+  username = '';
+  comment = '';
```
Comment (amejid) [inline src/modules/tests/commentInvolvement.js]: - [x] [OPTIONAL] The input fields are not clearing after the comment is submitted. This piece of code is setting the values to empty string but it's not manipulating the DOM. Kindly fix the issue for a better user experience. 🥸️

File: src/modules/movieCounter.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const movieCounter = (count) => {
+  const counter = document.getElementById('counter');
+  counter.innerText = count;
+  return count;
+};
```
Comment (amejid) [inline src/modules/movieCounter.js]: - [x] This function is not counting the number of items on the page. It is relying upon the data coming from the API and just updating the screen. According to the requirement, we should create a function that will count the number of items displayed on the page. 🤝️ One possible way to achieve this is by using the class name given for the individual movie `div`. I think you are using the class name `card` for an individual movie. 🧐️ You can use the output of the function to display the number.

**Example**
```
const movieCounter = () => {
  const movies = document.querySelectorAll('.card');
  return movies.length;
};
```

**Project requirements**
Screenshot from 2023-04-19 19-44-45

File: src/modules/counter.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+const counter = (item) => (item ? item.length : 0);
```
Comment (amejid) [inline src/modules/counter.js:1]: - [x] Please implement the comment counter in the way discussed earlier. 😎️

File: src/modules/tests/movieCounter.test.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @jest-environment jsdom
+ */
+import { movieCounter } from '../movieCounter.js';
+
+test('movieCounter function should count and display the number of movies correctly', () => {
```
Comment (amejid) [inline src/modules/tests/movieCounter.test.js]: - [ ] The test is not currently running. Please update the test to cover the soon-to-be-created movie counter function. Kindly try to cover as many edge cases as possible. One of the situations is when there are no movies displayed, the counter function should return 0. 🤓️
- [ ] Comment counter is not tested in the project. Please do the same type of tests for the comment counter too. 🤝️

File: src/modules/involvement.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable no-unused-vars */
```
Comment (amejid) [inline src/modules/involvement.js]: - [ ] Enforcing linters in our project helps us in writing consistent and clean code throughout the codebase. Kindly enable linter checks by removing this line. 🙂️

File: src/modules/movieCounter.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+
+/* eslint-disable import/prefer-default-export */
```
Comment (amejid) [inline src/modules/movieCounter.js]: - [ ] Same here. 🙂️

File: src/modules/movies.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  } catch (error) {
+    const displayMovies = document.querySelector('.display-container');
+    displayMovies.innerText = error;
+  }
+};
+
+/* eslint-disable import/prefer-default-export */
```
Comment (amejid) [inline src/modules/movies.js]: - [ ] Same here. 🙂️

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @Otegaa and @mnaibei, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Good UI design :heavy_check_mark: 
- Almost all functionalities working as expected :heavy_check_mark: 
- Good collaboration :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/movies.js (hunk 1 above)
Comment (mnaibei) [reply src/modules/movies.js:53]: Hi @amejid., the like counter does refresh as required after the first refresh.,

Comment (amejid) [reply src/modules/movies.js:53]: Hello @mnaibei, Could you please confirm with the project requirement document. Actually the text under Project requirements is a screenshot taken from the requirements as a reference.

Comment (mnaibei) [reply src/modules/movies.js:53]: @amejid Here is video evidence that the like count refreshes as required.

https://user-images.githubusercontent.com/69506217/233673914-12b6fdfd-0d7d-43a4-ada8-31255a0c0aa4.mov

Comment (amejid) [reply src/modules/movies.js:53]: @mnaibei Okay great. It was not working when I was testing things out earlier. It works now. The next reviewer will take this into account.

File: src/modules/counter.js (hunk 4 above)
Comment (Otegaa) [reply src/modules/counter.js:1]: My counter was called in pop-up.js and my data counter was not gotten from the API, It was gotten from the nodelist of ul, which is what was required.

File: src/modules/tests/movieCounter.test.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  if (movies.length > 999) {
+    return '999+';
+  }
+  return movies.length.toString();
+};
+
+describe('movieCounter function', () => {
+  it('should return "No movies" if movies array is empty', () => {
+    const result = movieCounter([]);
```
Comment (ghost) [inline src/modules/tests/movieCounter.test.js:17]: - [x] As requested earlier, I think it would be important to include tests for your `comments counter`  function to ensure that the app behaves as intended and provides accurate feedback to the user

File: src/modules/movies.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
```
Comment (ghost) [inline src/modules/movies.js]: - [x] as you have already solved the linter error, you can remove this line of code to keep your code base clean

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @mnaibei , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
✅ Correct use of Gitflow
✅ No linter error
✅ App works as expected


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:✔️

Hi @Team ✔️

Your project is Good to go! Great job 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request✔️
- Great design✔️
- works as expected✔️
- Great collaboration ✔️ 
- No linter errors✔️

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
